SU2_DOT/src/SU2_DOT.cpp

```diff
@@ -712,21 +710,13 @@ void SetProjection_AD(CGeometry *geometry, CConfig *config, CSurfaceMovement *su
   /*--- Register design variables as input and set them to zero
    * (since we want to have the derivative at alpha = 0, i.e. for the current design) ---*/
 
-
-
   for (iDV = 0; iDV < nDV; iDV++){
 
-    nDV_Value =  config->GetnDV_Value(iDV);
-
-    for (iDV_Value = 0; iDV_Value < nDV_Value; iDV_Value++){
+    for (iDV_Value = 0; iDV_Value < config->GetnDV_Value(iDV); iDV_Value++){
 
-      /*--- Initilization with su2double resets the index ---*/
+      config->SetDV_Value(iDV, iDV_Value, 0.0);
 
-      DV_Value = 0.0;
-
-      AD::RegisterInput(DV_Value);
-
-      config->SetDV_Value(iDV, iDV_Value, DV_Value);
+      AD::RegisterInput(config->GetDV_Value(iDV, iDV_Value));
```

Comment:
I wondered what made this break before and I can only think that before the local `DV_VALUE` was registered and below we now(!) extracted what is in `config->GetDV_Value(...)`. Before this PR it accessed the hidden `inputValues` so everything was good (no matter the input value to `SU2_TYPE::GetDerivative(...)`). 
```c++
      DV_Value = config->GetDV_Value(iDV, iDV_Value);
      my_Gradient = SU2_TYPE::GetDerivative(DV_Value);
```
Now you changed it to only work with the `DV_Value` that lives in the config and that is consistent with this change up here. 

The problem before this very commit was probably index management optimization because `config->SetDV_Value(iDV, iDV_Value, DV_Value);` before was just an assignment. But that is sth I would ask @jblueh in the Dev-meeting as well

Reply:
We were using always the same variable (declared at the top of the function).
On each loop iteration the =0 part would reset the previously registered index, then on extraction we were also writing over the same variable.